### PR TITLE
Some small performance improvements

### PR DESCRIPTION
**Pull request by neo1973**
## Description
Performance improvements by primarily avoiding copying around of data and allocations. Valgrinds DHAT tool was a great help to find those places. All in all nothing earth shattering but every little bit counts :slightly_smiling_face: 

## Motivation and context
faster == better!

## How has this been tested?
Checking in Valgrind that the hot sports went away and overall performance improves.

## What is the effect on users?
Maybe better performance on low end ARM devices, on your average desktop there probably wont be a noticeable difference.

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [X] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [X] All new and existing tests passed

### CODE REVIEW

**Comment by neo1973** [comment]
Jenkins build this please

**Comment by neo1973** [comment]
Jenkins build this please

File: xbmc/utils/Variant.h [hunk 1]
```diff
@@ -87,8 +87,10 @@ class CVariant
   uint64_t asUnsignedInteger(uint64_t fallback = 0u) const;
   uint32_t asUnsignedInteger32(uint32_t fallback = 0u) const;
   bool asBoolean(bool fallback = false) const;
-  std::string asString(const std::string &fallback = "") const;
-  std::wstring asWideString(const std::wstring &fallback = L"") const;
+  std::string asString(const std::string& fallback = "") const&;
```
Comment (howie-f) [inline xbmc/utils/Variant.h:90]: is this the same as `const std::string& asString(const std::string& fallback = "");` ?

Comment (howie-f) [reply xbmc/utils/Variant.h:90]: sorry, should have read the commit message :)

**Comment by neo1973** [comment]
Jenkins build this please

File: xbmc/guilib/DirtyRegionTracker.cpp [hunk 2]
```diff
@@ -77,12 +79,8 @@ CDirtyRegionList CDirtyRegionTracker::GetDirtyRegions()
...
-  while (i >= 0)
-	{
-    if (m_markedRegions[i].UpdateAge() >= buffering)
-      m_markedRegions.erase(m_markedRegions.begin() + i);
-
-    i--;
-  }
+  m_markedRegions.erase(
+      std::remove_if(m_markedRegions.begin(), m_markedRegions.end(),
+                     [buffering](CDirtyRegion& r) { return r.UpdateAge() > buffering; }),
```
Comment (popcornmix) [inline xbmc/guilib/DirtyRegionTracker.cpp:84]: Was it intentional to change the `>= buffering` to `> buffering`?
It looks like that could cause a change in behaviour.

Comment (neo1973) [reply xbmc/guilib/DirtyRegionTracker.cpp]: Nope, that wasn't intentional. Thanks for noticing :smile:

**Review by howie-f — APPROVED** [review]
lgtm and no runtime issues spotted

**Pull request merged**